pysd/tools/ncfiles.py

```diff
@@ -0,0 +1,412 @@
+"""
+Tools for importing and converting netCDF files generated from simulations run
+using PySD.
+"""
+import itertools
+import warnings
+
+from pathlib import Path
+from csv import QUOTE_NONE
+from typing import Union, Optional
+
+import xarray as xr
+import pandas as pd
+
+
+class NCFile():
+    """
+    Helper class to extract data from netCDF files.
+
+    Parameters
+    ----------
+    ncfile: str or pathlib.Path
+        Path to the netCDF file to process.
+    parallel: bool
+        When True, the Dataset is opened using chunks=-1 (see xarray
+        documentation for details) and DataArrays are processed in parallel
+        using dask delayed. Dask is not included as a requirement for pysd,
+        hence it must be installed separately. Setting parallel=True is
+        highly recommended when the Dataset contains large multidimensional
+        DataArrays.
+
+    """
+
+    valid_export_file_types = [".csv", ".tab"]
+
+    def __init__(self, filename: Union[str, Path],
+                 parallel: bool=False) -> None:
+
+        self.ncfile = NCFile._validate_nc_path(filename)
+        self.parallel = parallel
+        self.ds = self.open_nc()
+
+    def to_csv(self,
+               outfile: Optional[Union[str, Path]]="result.tab",
+               subset: Optional[list]=None,
+               time_in_row: Optional[bool]=False,
+               ) -> pd.DataFrame:
+        """
+        Convert netCDF file contents into comma or tab separated file.
+
+        Parameters
+        ----------
+        outfile: str or pathlib.Path
```

Comment:
add optional between parenthesis after the types for those arguments that are optional